### DISCUSSION TITLE
How can I build my project offline?

### DISCUSSION THREAD

**Discussion by anick3** [lumeland/lume#311] [General]
I use lume with the following plugins: esbuild, minify_html, multilanguage, sass, source_maps.

Is there a way to vendor (and if that's currently not possible,) to cache all dependencies locally, so I can build my project offline?

It currently says
```
TypeError: error sending request for url (https://deno.land/x/lume@v1.12.1/plugins/sass.ts): error trying to connect: dns error: failed to lookup address information: Temporary failure in name resolution
    at file:///home/anick/project/_config.ts:4:18
    at (file:///home/anick/project/_config.ts:4:18)
    | 3 | import multilanguage from "lume/plugins/multilanguage.ts";
    | 4 | import sass from "lume/plugins/sass.ts";
    | ~ | ~~~~~~~~~~~~~~~~~^
    | 5 | import source_maps from "lume/plugins/source_maps.ts";

    at async createSite (https://deno.land/x/lume@v1.12.1/cli/run.ts:51:17)
    at async build (https://deno.land/x/lume@v1.12.1/cli/build.ts:28:16)
    at async Command.execute (https://deno.land/x/cliffy@v0.25.2/command/command.ts:1463:7)
    at async (https://deno.land/x/lume@v1.12.1/cli.ts:147:3)
```
when I switch off my internet connection.

Btw. I love lume - it's so much better than everything I used before! Thank you so much for your work!

**Comment by oscarotero** [comment]
Deno cache all dependencies locally, so they are requested only the first time and then use the local cache. It should be enought to run Lume offline.

There's also the `deno vendor` command to store all dependencies locally in a directory (https://deno.land/manual@v1.27.2/tools/vendor). The problem is it doesn't work with `npm:` specifiers yet (it's in the [roadmap](https://github.com/denoland/deno/issues/15960) but I don't know if it will be implemented in the short term.

Lume had the `lume vendor` command that runs `deno vendor` internally with additional tweaks to make it work. But this was removed in lume 1.12.0 due problems with npm dependencies. If you really need this, you can downgrade to lume 1.11.4 and use it.

**Reply by anick3** [reply]
Hi, and thank you for the fast reponse!

I just tried it again - I ran `deno task build` and then I pulled out my network cable (:) ) and ran `deno task build` again. It says 

```
$ deno task build
Task build deno task lume
Task lume echo "import 'lume/cli.ts'" | deno run --unstable -A -
Loading config file file:///home/anick/project/_config.ts


TypeError: error sending request for url (https://wilsonl.in/minify-html/deno/0.10.1/index_bg.wasm): error trying to connect: dns error: failed to lookup address information: Temporary failure in name resolution
    at async createSite (https://deno.land/x/lume@v1.12.1/cli/run.ts:51:17)
    at async build (https://deno.land/x/lume@v1.12.1/cli/build.ts:28:16)
    at async Command.execute (https://deno.land/x/cliffy@v0.25.2/command/command.ts:1463:7)
    at async (https://deno.land/x/lume@v1.12.1/cli.ts:147:3)
```

My configuration is totally standard - exactly the same as the example described under https://lume.land/docs/overview/installation/.

**Reply by oscarotero** [reply]
Ok, it's because the minify_html plugin. It uses `minify-html` dependency (a wasm library) and it seems that wasm files are not cached by Deno. Can you try it without this plugin?

**Reply by anick3** [reply]
You are right, it works when I comment out the `minify-html` import and use lines! Thank you! I wonder, why I got that `sass` error the first time, though.
